### PR TITLE
*nix fixes

### PR DESCRIPTION
**Pull request by Yoplitein**
Fixes for #4.

### CODE REVIEW

File: src/gsrand.cpp [hunk 1]
```diff
@@ -381,7 +381,7 @@ void filter_default_content(vector<string>& unfiltered, const char ** default_li
 							vector<string>& search_paths, string ext, int& total_count, int& exclude_count, string& last_print)
 {
 	uint64 last_print_time = 0;
-	if (contentMode != CONTENT_EVERYTHING || maxContentBytes && search_paths.size() || skipUppercase)
+	if ((contentMode != CONTENT_EVERYTHING || maxContentBytes) && (search_paths.size() || skipUppercase))
```
Comment (wootguy) [inline src/gsrand.cpp]: Could you change it to this to preserve the original logic:
`(contentMode != CONTENT_EVERYTHING || (maxContentBytes && search_paths.size()) || skipUppercase)`

Everything else looks good.

**Comment by wootguy** [comment]
Cool, free code! Thanks for cleaning up the pile of whatsit in Util.

**Comment by Yoplitein** [comment]
My pleasure. :smile:

**Comment by incognico** [comment]
```
[ 40%] Building CXX object CMakeFiles/gsrand.dir/src/gsrand.cpp.o
/home/srcds/sc5/svencoop_addon/glsrc/Glitch-Life/src/gsrand.cpp: In function ‘void filter_default_content(std::vector<std::basic_string<char>
>&, const char**, int, std::vector<std::basic_string<char> >&, std::string, int&, int&, std::string&)’:
/home/srcds/sc5/svencoop_addon/glsrc/Glitch-Life/src/gsrand.cpp:417:11: error: use of deleted function ‘std::basic_ifstream<char>& std::basic_
ifstream<char>::operator=(const std::basic_ifstream<char>&)’
       fin = ifstream(search_paths[k] + unfiltered[i] + ext, ios::binary);
           ^
In file included from /home/srcds/sc5/svencoop_addon/glsrc/Glitch-Life/src/gsrand.h:6:0,
                 from /home/srcds/sc5/svencoop_addon/glsrc/Glitch-Life/src/gsrand.cpp:3:
/usr/include/c++/4.9/fstream:430:11: note: ‘std::basic_ifstream<char>& std::basic_ifstream<char>::operator=(const std::basic_ifstream<char>&)’
 is implicitly deleted because the default definition would be ill-formed:
     class basic_ifstream : public basic_istream<_CharT, _Traits>
           ^
/usr/include/c++/4.9/fstream:430:11: error: use of deleted function ‘std::basic_istream<char>& std::basic_istream<char>::operator=(const std::
basic_istream<char>&)’
In file included from /usr/include/c++/4.9/sstream:38:0,
                 from /home/srcds/sc5/svencoop_addon/glsrc/Glitch-Life/src/Util.h:4,
                 from /home/srcds/sc5/svencoop_addon/glsrc/Glitch-Life/src/gsrand.h:4,
                 from /home/srcds/sc5/svencoop_addon/glsrc/Glitch-Life/src/gsrand.cpp:3:
/usr/include/c++/4.9/istream:58:11: note: ‘std::basic_istream<char>& std::basic_istream<char>::operator=(const std::basic_istream<char>&)’ is
implicitly deleted because the default definition would be ill-formed:
     class basic_istream : virtual public basic_ios<_CharT, _Traits>
           ^
/usr/include/c++/4.9/istream:58:11: error: use of deleted function ‘std::basic_ios<char>& std::basic_ios<char>::operator=(const std::basic_ios
<char>&)’
In file included from /usr/include/c++/4.9/ios:44:0,
                 from /usr/include/c++/4.9/istream:38,
                 from /usr/include/c++/4.9/sstream:38,
                 from /home/srcds/sc5/svencoop_addon/glsrc/Glitch-Life/src/Util.h:4,
                 from /home/srcds/sc5/svencoop_addon/glsrc/Glitch-Life/src/gsrand.h:4,
                 from /home/srcds/sc5/svencoop_addon/glsrc/Glitch-Life/src/gsrand.cpp:3:
/usr/include/c++/4.9/bits/basic_ios.h:66:11: note: ‘std::basic_ios<char>& std::basic_ios<char>::operator=(const std::basic_ios<char>&[22/1833]
licitly deleted because the default definition would be ill-formed:
     class basic_ios : public ios_base
           ^
In file included from /usr/include/c++/4.9/ios:42:0,
                 from /usr/include/c++/4.9/istream:38,
                 from /usr/include/c++/4.9/sstream:38,
                 from /home/srcds/sc5/svencoop_addon/glsrc/Glitch-Life/src/Util.h:4,
                 from /home/srcds/sc5/svencoop_addon/glsrc/Glitch-Life/src/gsrand.h:4,
                 from /home/srcds/sc5/svencoop_addon/glsrc/Glitch-Life/src/gsrand.cpp:3:
/usr/include/c++/4.9/bits/ios_base.h:789:5: error: ‘std::ios_base& std::ios_base::operator=(const std::ios_base&)’ is private
     operator=(const ios_base&);
     ^
In file included from /usr/include/c++/4.9/ios:44:0,
                 from /usr/include/c++/4.9/istream:38,
                 from /usr/include/c++/4.9/sstream:38,
                 from /home/srcds/sc5/svencoop_addon/glsrc/Glitch-Life/src/Util.h:4,
                 from /home/srcds/sc5/svencoop_addon/glsrc/Glitch-Life/src/gsrand.h:4,
                 from /home/srcds/sc5/svencoop_addon/glsrc/Glitch-Life/src/gsrand.cpp:3:
/usr/include/c++/4.9/bits/basic_ios.h:66:11: error: within this context
     class basic_ios : public ios_base
           ^
In file included from /home/srcds/sc5/svencoop_addon/glsrc/Glitch-Life/src/gsrand.h:6:0,
                 from /home/srcds/sc5/svencoop_addon/glsrc/Glitch-Life/src/gsrand.cpp:3:
/usr/include/c++/4.9/fstream:430:11: error: use of deleted function ‘std::basic_filebuf<char>& std::basic_filebuf<char>::operator=(const std::
basic_filebuf<char>&)’
     class basic_ifstream : public basic_istream<_CharT, _Traits>
           ^
/usr/include/c++/4.9/fstream:72:11: note: ‘std::basic_filebuf<char>& std::basic_filebuf<char>::operator=(const std::basic_filebuf<char>&)’ is
implicitly deleted because the default definition would be ill-formed:
     class basic_filebuf : public basic_streambuf<_CharT, _Traits>
           ^
In file included from /usr/include/c++/4.9/ios:43:0,
                 from /usr/include/c++/4.9/istream:38,
                 from /usr/include/c++/4.9/sstream:38,
                 from /home/srcds/sc5/svencoop_addon/glsrc/Glitch-Life/src/Util.h:4,
                 from /home/srcds/sc5/svencoop_addon/glsrc/Glitch-Life/src/gsrand.h:4,
                 from /home/srcds/sc5/svencoop_addon/glsrc/Glitch-Life/src/gsrand.cpp:3:
/usr/include/c++/4.9/streambuf:810:7: error: ‘std::basic_streambuf<_CharT, _Traits>& std::basic_streambuf<_CharT, _Traits>::operator=(const st
d::basic_streambuf<_CharT, _Traits>&) [with _CharT = char; _Traits = std::char_traits<char>]’ is private
       operator=(const basic_streambuf&) { return *this; };
       ^
In file included from /home/srcds/sc5/svencoop_addon/glsrc/Glitch-Life/src/gsrand.h:6:0,
                 from /home/srcds/sc5/svencoop_addon/glsrc/Glitch-Life/src/gsrand.cpp:3:
/usr/include/c++/4.9/fstream:72:11: error: within this context
     class basic_filebuf : public basic_streambuf<_CharT, _Traits>
           ^
CMakeFiles/gsrand.dir/build.make:123: recipe for target 'CMakeFiles/gsrand.dir/src/gsrand.cpp.o' failed
make[2]: *** [CMakeFiles/gsrand.dir/src/gsrand.cpp.o] Error 1
CMakeFiles/Makefile2:60: recipe for target 'CMakeFiles/gsrand.dir/all' failed
make[1]: *** [CMakeFiles/gsrand.dir/all] Error 2
Makefile:76: recipe for target 'all' failed
make: *** [all] Error 2
```
maybe it is the time to hang myself

**Comment by Yoplitein** [comment]
I don't even know what that error means. Try a newer version of gcc or clang?

**Comment by incognico** [comment]
Im using the latest debian version like 8 or something with the default
build-essentials and cmake. I'm not at the pc right now but I guess debian
still uses gcc instead of clang because it is known to be and use rotten
software.

**Comment by Yoplitein** [comment]
Debian and its packages are often fabulously out of date. Judging from the include paths in the error log, you're two major gcc versions behind. You might be able to get gcc 6 from the sid repos.

How is clang rotten software?

**Comment by incognico** [comment]
Of course by rotten I mean debian and its anicent package versions, not clang.

It now compiled fine with clang and libstdc++-6 from sid.

Thanks for your updates.

**Pull request merged**